response.go

```diff
@@ -0,0 +1,248 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tspclient
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/notaryproject/tspclient-go/pki"
+)
+
+// Response is a time-stamping response.
+//
+//	TimeStampResp ::= SEQUENCE {
+//	 status          PKIStatusInfo,
+//	 timeStampToken  TimeStampToken  OPTIONAL }
+type Response struct {
+	Status         pki.StatusInfo
+	TimeStampToken asn1.RawValue `asn1:"optional"`
+}
+
+// signingCertificate contains certificate hash and identifier of the
+// TSA signing certificate.
+//
+// Reference: RFC 2634 5.4 signingCertificate
+//
+//	signingCertificate ::=  SEQUENCE {
+//	 certs        SEQUENCE OF ESSCertID,
+//	 policies     SEQUENCE OF PolicyInformation OPTIONAL }
+type signingCertificate struct {
+	// Certificates contains the list of certificates. The first certificate
+	// MUST be the signing certificate used to verify the timestamp token.
+	Certificates []eSSCertID
+
+	// Policies suggests policy values to be used in the certification path
+	// validation.
+	Policies asn1.RawValue `asn1:"optional"`
+}
+
+// signingCertificateV2 contains certificate hash and identifier of the
+// TSA signing certificate.
+//
+// Reference: RFC 5035 3 signingCertificateV2
+//
+//	signingCertificateV2 ::=  SEQUENCE {
+//	 certs        SEQUENCE OF ESSCertIDv2,
+//	 policies     SEQUENCE OF PolicyInformation OPTIONAL }
+type signingCertificateV2 struct {
+	// Certificates contains the list of certificates. The first certificate
+	// MUST be the signing certificate used to verify the timestamp token.
+	Certificates []eSSCertIDv2
+
+	// Policies suggests policy values to be used in the certification path
+	// validation.
+	Policies asn1.RawValue `asn1:"optional"`
+}
+
+// eSSCertID uniquely identifies a certificate.
+//
+// Reference: RFC 2634 5.4.1
+//
+//	eSSCertID ::=  SEQUENCE {
+//	 certHash                 Hash,
+//	 issuerSerial             IssuerSerial OPTIONAL }
+type eSSCertID struct {
+	// CertHash is the certificate hash using algorithm SHA1.
+	// It is computed over the entire DER-encoded certificate
+	// (including the signature)
+	CertHash []byte
+
+	// IssuerSerial holds the issuer and serialNumber of the certificate.
+	// When it is not present, the SignerIdentifier field in the SignerInfo
+	// will be used.
+	IssuerSerial issuerAndSerial `asn1:"optional"`
+}
+
+// eSSCertIDv2 uniquely identifies a certificate.
+//
+// Reference: RFC 5035 4
+//
+//	eSSCertIDv2 ::=  SEQUENCE {
+//	 hashAlgorithm           AlgorithmIdentifier
+//	 	DEFAULT {algorithm id-sha256},
+//	 certHash                 Hash,
+//	 issuerSerial             IssuerSerial OPTIONAL }
+type eSSCertIDv2 struct {
+	// HashAlgorithm is the hashing algorithm used to hash certificate.
+	// When it is not present, the default value is SHA256 (id-sha256).
+	// Supported values are SHA256, SHA384, and SHA512
```

Comment:
Are we validating this?

Reply:
Yes, it is validated. But it's not a part of `Response validation`, it's a crucial step of `TimestampToken verification`, where you can find it in `token.go`.